guide/13-managing-arcgis-applications/hub-basic-guide.ipynb

```diff
@@ -0,0 +1,673 @@
+     "output_type": "stream",
+     "text": [
+      "Enter password: ········\n"
+     ]
+    }
+   ],
+   "source": [
+    "gis1 = GIS(\"https://www.arcgis.com\", \"MMajumdar_geosaurus\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bb4272f9",
+   "metadata": {},
+   "source": [
+    "We will now connect to the Hub associated with this Web GIS."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "654623b8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hub_basic = gis1.hub"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3ca6351e",
+   "metadata": {},
+   "source": [
+    "### Adding new sites and pages\n",
+    "\n",
+    "ArcGIS Hub allows you to build configurable and branded websites for sharing internal and public content, such as apps, data, and media. If you're using ArcGIS Hub Premium, you can launch a site by creating an __initiative__ (insert premium hub link here). You can build a Hub Basic Site using the ArcGIS API for Python as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "99e27040",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div class=\"item_container\" style=\"height: auto; overflow: hidden; border: 1px solid #cfcfcf; border-radius: 2px; background: #f6fafa; line-height: 1.21429em; padding: 10px;\">\n",
+       "                    <div class=\"item_left\" style=\"width: 210px; float: left;\">\n",
+       "                       <a href='https://geosaurus.maps.arcgis.com/home/item.html?id=0ddf92bbd174426b8c068b71d9758413' target='_blank'>\n",
+       "                        <img src='http://static.arcgis.com/images/desktopapp.png' class=\"itemThumbnail\">\n",
+       "                       </a>\n",
+       "                    </div>\n",
+       "\n",
+       "                    <div class=\"item_right\"     style=\"float: none; width: auto; overflow: hidden;\">\n",
+       "                        <a href='https://geosaurus.maps.arcgis.com/home/item.html?id=0ddf92bbd174426b8c068b71d9758413' target='_blank'><b>Python Site</b>\n",
+       "                        </a>\n",
+       "                        <br/><img src='https://geosaurus.maps.arcgis.com/home/js/jsapi/esri/css/images/item_type_icons/layers16.png' style=\"vertical-align:middle;\">Hub Site Application by MMajumdar_geosaurus\n",
+       "                        <br/>Last Modified: December 12, 2022\n",
+       "                        <br/>0 comments, 0 views\n",
+       "                    </div>\n",
+       "                </div>\n",
+       "                "
+      ],
+      "text/plain": [
+       "<Item title:\"Python Site\" type:Hub Site Application owner:MMajumdar_geosaurus>"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "site1 = hub_basic.sites.add('Python Site')\n",
+    "site1.item"
+   ]
```

Comment:
* When using an environment built from the master branch of the src repo, the `add()` method returns:
```pytb
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 site1 = hub_basic.sites.add('Python Site6')
      2 site1.item

File ~/Job/repos/geosaurus/src/arcgis/apps/hub/sites.py:1058, in SiteManager.add(self, title, subdomain)
   1055     _site_data = json.load(f)
   1057 # Register site and update its data
-> 1058 _data = self._create_and_register_site(
   1059     site, subdomain, _site_data, content_group_id, collab_group_id
   1060 )
   1061 _data = json.dumps(_data)
   1062 site.update(item_properties={"text": _data, "url": domain})

File ~/Job/repos/geosaurus/src/arcgis/apps/hub/sites.py:698, in SiteManager._create_and_register_site(self, site, subdomain, site_data, content_group_id, collab_group_id)
    694 basemap = {}
    696 if self._gis._portal.is_arcgisonline:
    697     # register site as an app
--> 698     _app_dict = site.register(app_type="browser", redirect_uris=[site.url])
    699     client_key = _app_dict["client_id"]
    701     # Check for length of domain

File ~/Job/repos/geosaurus/src/arcgis/gis/__init__.py:16688, in Item.register(self, app_type, redirect_uris, http_referers, privileges)
  16686     params["privileges"] = privileges
  16687 url = "%soauth2/registerApp" % self._portal.resturl
> 16688 res = self._portal.con.post(url, params)
  16689 self._hydrated = False
  16690 self._hydrate()

File ~/Job/repos/geosaurus/src/arcgis/gis/_impl/_con/_connection.py:1522, in Connection.post(self, path, params, files, **kwargs)
   1520 if return_raw_response:
   1521     return resp
-> 1522 return self._handle_response(
   1523     resp=resp,
   1524     out_path=out_path,
   1525     file_name=file_name,
   1526     try_json=try_json,
   1527     force_bytes=kwargs.pop("force_bytes", False),
   1528 )

File ~/Job/repos/geosaurus/src/arcgis/gis/_impl/_con/_connection.py:998, in Connection._handle_response(self, resp, file_name, out_path, try_json, force_bytes, ignore_error_key)
    996             return data
    997         errorcode = data["error"]["code"] if "code" in data["error"] else 0
--> 998         self._handle_json_error(data["error"], errorcode)
    999     return data
   1000 else:

File ~/Job/repos/geosaurus/src/arcgis/gis/_impl/_con/_connection.py:1021, in Connection._handle_json_error(self, error, errorcode)
   1018                 # _log.error(errordetail)
   1020 errormessage = errormessage + "\n(Error Code: " + str(errorcode) + ")"
-> 1021 raise Exception(errormessage)

Exception: Request signature does not match. The request has to be signed.
(Error Code: 400)
```

> **NOTE:**  this line runs in an environment built from the changes submitted in src #PR 10145